### PR TITLE
linode-cli: init at 1.4.7

### PR DESCRIPTION
**Pull request by nixy**
###### Motivation for this change

Want to make the Linode-CLI available through Nix, because Nix is the best :) 

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  # Has no docs output
+  outputs = [ "out" ];
+
+  # Has no tests
+  doCheck = false;
+
+  src = fetchgit {
+    url = "git://github.com/linode/cli";
+    rev = "8d475f3ff161b64dbe1aff167897974f92475a3c";
+    sha256 = "1c24c7n3fq9c3yibw1i4zlk3hdcjaqcd5cq4kvvw4hfg2l3w6a21";
```
Comment (Mic92) [inline pkgs/tools/virtualization/linode-cli/default.nix]: Please use `fetchFromGitHub` here instead of `fetchgit`

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  # Has no docs output
+  outputs = [ "out" ];
+
+  # Has no tests
+  doCheck = false;
+
+  src = fetchgit {
+    url = "git://github.com/linode/cli";
+    rev = "8d475f3ff161b64dbe1aff167897974f92475a3c";
```
Comment (Mic92) [inline pkgs/tools/virtualization/linode-cli/default.nix]: Please use the version as revision here: https://github.com/linode/cli/releases

```
  rev = "v${version}";
```

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+buildPerlPackage rec {
+  name = "linode-cli-${version}";
+  version = "1.4.7";
+
+  # Has no docs output
+  outputs = [ "out" ];
+
+  # Has no tests
+  doCheck = false;
```
Comment (Mic92) [inline pkgs/tools/virtualization/linode-cli/default.nix]: The package sill misses dependencies: 

```
Warning: prerequisite JSON 0 not found.
Warning: prerequisite LWP::UserAgent 0 not found.
Warning: prerequisite Mozilla::CA 0 not found.
Warning: prerequisite Try::Tiny 0 not found.
Warning: prerequisite WebService::Linode 0 not found.
```

**Comment by nixy** [comment]
So it looks like my initial attempt at packaging this was pretty naive, but the binary worked surprisingly well all things considered.

I have included all the libraries and am using fetchFromGitHub now, but I am a bit stumped on a library path issue. When I try and run some of the executables I get the following:

Can't locate Linode/CLI.pm in `@INC` (you may need to install the Linode::CLI module) (`@INC` contains: /nix/store/lxyzi71fzqy7av2fs4n65286cbiibwcd-perl-linode-cli-1.4.7/bin/lib /nix/store/8svnbpw903za1imxn73fxyjql6hvviz6-perl-5.22.2/lib/perl5/site_perl/5.22.2/darwin-2level /nix/store/8svnbpw903za1imxn73fxyjql6hvviz6-perl-5.22.2/lib/perl5/site_perl/5.22.2 /nix/store/8svnbpw903za1imxn73fxyjql6hvviz6-perl-5.22.2/lib/perl5/5.22.2/darwin-2level /nix/store/8svnbpw903za1imxn73fxyjql6hvviz6-perl-5.22.2/lib/perl5/5.22.2 .) at result/bin/linode-account line 13.
BEGIN failed--compilation aborted at result/bin/linode-account line 13.

I am not terribly familiar with Perl so I am not sure how the `@INC` variable is defined, or how to modify it in a Nix expression. Any advice? I have taken a look through some other Perl packages and tried modifying `$PERL5LIB` but it didn't seem to have any effect.

**Comment by joachifm** [comment]
You might need to wrap the executable, using some variation of
```
nativeBuildInputs = [ makeWrapper ];
# ...
postInstall = ''
   # ...
  wrapProgram $out/bin/foo --prefix PERL5LIB : "$PERL5LIB"
'';
```

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  version = "1.4.7";
+
+  src = fetchFromGitHub {
+    owner = "linode";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "1wiz067wgxi4z4rz4n9p7dlvx5z4hkl2nxpfvhikl6dri4m2nkkp";
+  };
+
+  # src = fetchgit {
```
Comment (joachifm) [inline pkgs/tools/virtualization/linode-cli/default.nix]: Please remove this

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  #   url = "git://github.com/linode/cli";
+  #   rev = "8d475f3ff161b64dbe1aff167897974f92475a3c"; 
+  #   sha256 = "1c24c7n3fq9c3yibw1i4zlk3hdcjaqcd5cq4kvvw4hfg2l3w6a21";
+  # };
+
+  # Has no docs output
+  outputs = [ "out" ];
```
Comment (joachifm) [inline pkgs/tools/virtualization/linode-cli/default.nix]: This is the default

File: pkgs/top-level/perl-packages.nix [hunk 6]
```diff
@@ -14958,4 +14958,19 @@ let self = _self // overrides; _self = with self; {
     };
   };
 
+  WebServiceLinode = buildPerlModule rec {
```
Comment (Mic92) [inline pkgs/top-level/perl-packages.nix]: Please do this in a separate commit with the message: `perlPackages.WebServiceLinode: init at 0.28`

File: pkgs/tools/virtualization/linode-cli/default.nix (hunk 5 above)
Comment (nixy) [inline pkgs/tools/virtualization/linode-cli/default.nix]: The default outputs appears to contain "devdoc" at well which caused this build to fail.

Comment (joachifm) [inline pkgs/tools/virtualization/linode-cli/default.nix]: Ah, that must be a perl thing

**Comment by nixy** [comment]
I believe this should be good now. Thanks a bunch ya'll. Learning about all kinds of fun Nix functions that make life easier.

**Comment by nixy** [comment]
Anything else I need to do here?

**Comment by joachifm** [comment]
The travis build fails, can you investigate? Other than that it also needs a rebase.

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    for n in "$out/bin"/*; do
+      wrapProgram "$n" --prefix PERL5LIB : "$PERL5LIB"
+    done
+  '';
+
+  # Has no tests
+  doCheck = false;
+
+  # # Has no docs output
+  # outputs = [ "out" ];
```
Comment (7c6f434c) [inline pkgs/tools/virtualization/linode-cli/default.nix]: I think Perl packages are expected to have `devdoc` output, but there is none in this case, which causes the failure.

**Comment by nixy** [comment]
I made the changes suggested and I believe the travis build should run, but it looks like Travis rejected my build. It looks like it gave the following errors:

```
abuse detected: user page on GitHub gives 404
abuse detected: known offender
```

Anyone know what I can do to prevent that?

**Comment by rvl** [comment]
Thanks @nixy @Mic92 @7c6f434c
